### DISCUSSION TITLE
Any way to iterate over a list?

### DISCUSSION THREAD

**Discussion by wilg** [inkle/ink#828] [General]
I have a list of "cases" (missions) that can be accepted or not. When they are accepted, I'd like their choices to be able to be added to the world.

```ink
LIST cases = SeabirdSoiree, FogGully, HouseParty
VAR availableCases = ()
VAR acceptedCases = ()

=== Free_Roam
<- Case_SeabirdSoiree.Interactions
<- Case_FogGully.Interactions
<- Case_HouseParty.Interactions
+ [Interact: Phone ((id: phone))]
    -> Phone_Options
- -> Free_Roam
```

Ideally, here, I'd like to automatically include the interactions from each of the `acceptedCases`. But I think there is no way to iterate over a list? Or perhaps there is a better way to structure this.

**Comment by joningold** [comment]
you can iterate over a list by making a loop: think of it like a deck of cards, and then deal them out one card at a time.

```

// copy the list, so we can work through it
~ temp loopList = TheListVariableYouWantToIterate 
- (loop)
~ temp element = LIST_MIN(loopList)
~ loopList -= element
 {element:
  // do whatever you want to do with the element 
 }
 { loopList:
   // still more elements so get the next one
    -> loop 
   }
- (loopisover)
```

**Reply by wilg** [reply]
Whoa, neat! Thinking about if it can be wrapped up into a loop function...

**Reply by joningold** [reply]
it can, though with functions you want to recurse instead of divert, but the idea is the same.

**Reply by joethephish** [reply]
You could wrap it up into a tunnel structure:

```
LIST Fruit = Apple, Banana, Pear


-> each(LIST_ALL(Fruit), -> printFruit) ->

= printFruit(fruit)
You {|also} have a {fruit}. 
->->



=== each(list, -> do)
// copy the list, so we can work through it
~ temp loopList = list 
- (top)
~ temp element = LIST_MIN(loopList)
~ loopList -= element
{element:
    -> do(element) ->
}
{ loopList:
    // still more elements so get the next one
    -> top 
}
->->
```
